pkg/cloudprovider/aws/instance.go

```diff
@@ -143,6 +197,13 @@ func (p *InstanceProvider) launchInstances(ctx context.Context, constraints *v1a
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
+	if len(createFleetOutput.Errors) > 0 {
```

Comment:
You can skip this check, since the for loop will not execute if the length == 0.